app/controllers/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+class UsersController < ApplicationController
+  def index
+    @users = User.all
+    render json: @users
+  end
+
+  def show
+    if @user = User.find_by(id: params[:id])
+      render json: @user
+    else
+      render json: 'User not found'
+    end
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      render json: 'User created successfully'
+    else
+      render json: 'User creation failed'
+    end
+  end
+
+  def new
+    @user = User.new
+    render json: 'In new method'
+  end
+
+  def update
+    @user = User.find_by(id: params[:id])
+    if @user
+      if @user.update(user_params)
+        render json: @user
+      else
+        render json: 'Update unsuccessful'
+      end
+    else
+      render json: 'User not found'
+    end
+  end
+
```

Comment:
Can be optimised as:
Read about "&" operator
```suggestion
def update
  @user = User.find_by(id: params[:id])
  if user&.update(user_params)
    render json: user
  else
    render json: user ? 'Update unsuccessful' : 'User not found'
  end
end
```

Reply:
Updated.